File: 83-remove-duplicates-from-sorted-list/83-remove-duplicates-from-sorted-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
         ListNode *sentinel = new ListNode(INT_MIN, head);
         ListNode *cur = head;
         ListNode *prev = sentinel;
        
        while ( cur ) {
            if (cur->val == prev->val) {
                ListNode *tmp = cur;
                prev->next = cur->next;
                cur = cur->next;
                delete tmp;
            }
            else {
                prev = cur;
                cur = cur->next;
            }
        }
        
        head = sentinel->next;
        delete sentinel;
        return head;
    }
    
};

File: 83-remove-duplicates-from-sorted-list/83-remove-duplicates-from-sorted-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def deleteDuplicates(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        sentinel = ListNode( float("-inf"), head)
        
        curr = head
        prev = sentinel
        
        while curr is not None:
            if prev.val == curr.val:
                prev.next = curr.next
                curr = curr.next
            else:
                prev = curr
                curr = curr.next
        
        return sentinel.next
        
        